src/main/java/io/cdap/plugin/salesforce/connector/SalesforceConnector.java

```diff
@@ -0,0 +1,188 @@
+    AuthenticatorCredentials credentials = new AuthenticatorCredentials(config.getUsername(), config.getPassword(),
+                                                                        config.getConsumerKey(),
+                                                                        config.getConsumerSecret(),
+                                                                        config.getLoginUrl());
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    int count = 0;
+    try {
+      PartnerConnection partnerConnection = SalesforceConnectionUtil.getPartnerConnection(credentials);
+      DescribeGlobalResult dgr = partnerConnection.describeGlobal();
+      // Loop through the array to get all the objects.
+      for (int i = 0; i < dgr.getSobjects().length; i++) {
+        String name = dgr.getSobjects()[i].getName();
+        String label = dgr.getSobjects()[i].getLabel();
+        BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_OBJECTS).
+          canBrowse(false).canSample(true));
+        entity.addProperty(LABEL_NAME, BrowseEntityPropertyValue.builder(label, BrowseEntityPropertyValue.
+          PropertyType.STRING).build());
+        browseDetailBuilder.addEntity(entity.build());
+        count++;
+      }
+    } catch (ConnectionException e) {
+      throw  new IOException("Unable to create the connection.", e);
+    }
+    return browseDetailBuilder.setTotalCount(count).build();
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+    throws IOException {
+    ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+    Map<String, String> properties = new HashMap<>();
+    properties.put(io.cdap.plugin.common.ConfigUtil.NAME_USE_CONNECTION, "true");
+    properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+    String tableName = connectorSpecRequest.getPath();
+    if (tableName != null) {
+      properties.put(SalesforceSourceConstants.PROPERTY_SOBJECT_NAME, tableName);
+      properties.put(SalesforceSinkConfig.PROPERTY_SOBJECT, tableName);
+    }
+    AuthenticatorCredentials authenticatorCredentials = config.getAuthenticatorCredentials();
+    try {
+      SObjectDescriptor sObjectDescriptor = SObjectDescriptor.fromName(tableName, authenticatorCredentials);
+      Schema schema = SalesforceSchemaUtil.getSchema(authenticatorCredentials, sObjectDescriptor);
+      specBuilder.setSchema(schema);
+    } catch (ConnectionException e) {
+      throw new IOException("Unable to generate Schema", e);
+    }
+    return specBuilder.addRelatedPlugin(new PluginSpec(SalesforceBatchSource.NAME, BatchSource.PLUGIN_TYPE,
+                                                       properties)).
+      addRelatedPlugin(new PluginSpec(SalesforceBatchSink.PLUGIN_NAME, BatchSink.PLUGIN_TYPE, properties)).build();
```

Comment:
streaming source?

Reply:
We are supporting browse functionality to browse objects only which is used in source and sink only.